File: smartadapter/src/test/java/smartadapter/SmartEndlessScrollRecyclerAdapterTest.java
package smartadapter;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertNotNull;

/*
 * Created by Manne Öhlund on 2019-07-28.
 * Copyright (c) All rights reserved.
 */

@RunWith(RobolectricTestRunner.class)
public class SmartEndlessScrollRecyclerAdapterTest {

    @Test
    public void testImplicitInstantiate() {
        SmartEndlessScrollRecyclerAdapter.Companion.empty().create();
    }

    @Test
    public void testExplicitInstantiate() {
        SmartEndlessScrollRecyclerAdapter smartRecyclerAdapter = SmartEndlessScrollRecyclerAdapter.Companion.empty().create();
        assertNotNull(smartRecyclerAdapter);
    }
}

File: smartadapter/src/test/java/smartadapter/SmartEndlessScrollRecyclerAdapterTest.kt
package smartadapter

import org.junit.Assert.assertNotNull
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

/*
 * Created by Manne Öhlund on 2019-07-28.
 * Copyright (c) All rights reserved.
 */

@RunWith(RobolectricTestRunner::class)
class SmartEndlessScrollRecyclerAdapterTest {

    @Test
    fun testImplicitInstantiate() {
        val smartRecyclerAdapter = SmartEndlessScrollRecyclerAdapter.empty().create<SmartEndlessScrollRecyclerAdapter>()
        assertNotNull(smartRecyclerAdapter)
    }

    @Test
    fun testExplicitInstantiate() {
        val smartRecyclerAdapter = SmartEndlessScrollRecyclerAdapter.empty().create<SmartEndlessScrollRecyclerAdapter>()
        assertNotNull(smartRecyclerAdapter)
    }
}